### DISCUSSION TITLE
Can mike be used without GitHub?

### DISCUSSION THREAD

**Discussion by tvdboom** [jimporter/mike#222] [Q&A] [answered]
Hi,

I have an mkdocs documentation page on a devops git repo. I am building the page on a ci pipeline and deploying it to an azure webapp using a docker container. Is it possible to make `mike` work with this setup?

Ideally, I would be able to use `mike deploy` to push the website to another branch in the same devops repo, and read the files from there using docker, but I am open for other solutions.

Thanks

**Comment by jimporter** [comment, marked as answer]
> Ideally, I would be able to use `mike deploy` to push the website to another branch in the same devops repo, and read the files from there using docker, but I am open for other solutions.

That's essentially the only thing mike does in the first place, so you're probably good. Every time you run `mike deploy` (or some other command to modify your generated docs), it adds a new commit to `gh-pages` so that the latest commit on that branch is your full site. From there, you can do whatever your other tooling needs.
